lib/safe-pg-migrations/plugins/idempotent_statements.rb

```diff
@@ -15,42 +14,44 @@ module IdempotentStatements
         return
       end
 
-      remove_index(table_name, name: index_definition.name)
+      remove_index(table_name, column_name, **options)
       super
     end
 
-    ruby2_keywords def add_column(table_name, column_name, type, *)
+    def add_column(table_name, column_name, type, **options)
       if column_exists?(table_name, column_name) && !column_exists?(table_name, column_name, type)
         error_message = "/!\\ Column '#{column_name}' already exists in '#{table_name}' with a different type"
         raise error_message
       end
 
-      return super unless column_exists?(table_name, column_name, type)
+      options_without_default_value_backfill = options.except(:default_value_backfill)
 
-      log_message(<<~MESSAGE.squish
-        /!\\ Column '#{column_name}' already exists in '#{table_name}' with the same type (#{type}).
-        Skipping statement.
-      MESSAGE
-                 )
+      if column_exists?(table_name, column_name, type)
+        log_message(<<~MESSAGE.squish
+          /!\\ Column '#{column_name}' already exists in '#{table_name}' with the same type (#{type}).
+          Skipping statement.
+        MESSAGE
+                   )
+      else
+        super(table_name, column_name, type, **options_without_default_value_backfill)
+      end
```

Comment:
Nitpick: I think you should keep the initial order if possible, to help the review of the PR and focus on mandatory changes:
```
options_without_default_value_backfill = options.except(:default_value_backfill)

return super(table_name, column_name, type, **options_without_default_value_backfill) unless column_exists?(table_name, column_name, type)

log_message(<<~MESSAGE.squish
          /!\\ Column '#{column_name}' already exists in '#{table_name}' with the same type (#{type}).
          Skipping statement.
        MESSAGE
                   )

Reply:
I prefer to keep it that way because I prefer if instead of unless